builder/proxmox/common/step_start_vm.go

```diff
@@ -17,6 +17,10 @@ import (
 // in API calls.
 type stepStartVM struct{}
```

Comment:
Hi, weighing in here because I saw the conversation about the statebag and the proxmoxVMcreator -- this is definitely a mild abuse of the state bag. I only did a light read of the code, but can you add ProxmoxVMCreator to the step struct itself, here? 

```suggestion
type stepStartVM struct{
    vmCreator ProxmoxVMCreator
}
```

Reply:
Oh nice, yeah, it works by way of leaning on the interface, no statebag hacks required. I've updated this PR.